packages/libs/error-stack/src/report.rs

```diff
@@ -617,10 +665,22 @@ impl<Context> fmt::Debug for Report<Context> {
     }
 }
 
+#[cfg(feature = "std")]
+impl<Context> std::process::Termination for Report<Context>
+where
+    Context: fmt::Debug,
+{
+    fn report(self) -> std::process::ExitCode {
+        std::process::ExitCode::from(self.inner.exit_code.unwrap_or(1))
+    }
+}
+
```

Comment:
Instead of returning the exit code from the Report, it's probably more intuitive, if we request it:
```rust
#[cfg(feature = "std")]
impl<Context> std::process::Termination for Report<Context>
    where
        Context: fmt::Debug,
{
    fn report(self) -> ExitCode {
        #[cfg(not(nightly))]
        return ExitCode::FAILURE;
        
        #[cfg(nightly)]
        self.request_ref::<ExitCode>()
            .next()
            .copied()
            .unwrap_or(ExitCode::FAILURE)
    }
}
```

So your example would become
```rust
Report::new(CustomError)
    .attach(ExitCode::from(24))
    .attach_printable("A custom exit code!"))
```

Reply:
Makes sense to me!

One question though do you think whether there might be any merit in the method `attach_exit_code(self, exit_code: u8) -> Self` such that a user doesn't have to import `ExitCode` every time they wish to give an exit code? It would just internally call `attach` and generate an `ExitCode` from the given u8. I could see this being a nice QoL.

Reply:
I hear you and I see the QoL by just calling `attach_exit_code`. However, this should also be added to `ResultExt` and `FutureExt` then (and later on `IteratorExt` and `StreamExt` when implemented). This would add 5 more methods (a `_lazy` variant would most probably not be needed), which I think are pretty many for only a small QoL. As long as it is not mandatory, I would like to stick to `attach`. 🙂 

Too bad, that there is no `impl Termination for Result<(), E> for E: Termination`, so `Result<(), Report<_>>::report()` will still return `ExitCode::FAILURE`. 😦  (It's using `Result<!, _>`)

Reply:
Fair point a user could just define a closure that does what that function does and avoid importing the `ExitCode`.

I agree on the lacking impl for `Termination` would be nice. I'm not sure if they intend to add that in the future, but it would make sense if they do.